tkp/db/quality.py

```diff
@@ -2,78 +2,79 @@
 check image quality
 """
 import logging
-from collections import namedtuple
-import tkp.db
+from tkp.db.model import Rejectreason, Rejection
 
 logger = logging.getLogger(__name__)
 
-# TODO: need to think of a way to sync this with tkp/db/tables/rejection.sql
-RejectReason = namedtuple('RejectReason', 'id desc')
-
-reason = {
-    'rms': RejectReason(id=0, desc='RMS invalid'),
-    'beam': RejectReason(id=1, desc='beam invalid'),
-    'bright_source': RejectReason(id=2, desc='bright source near'),
-    'tau_time': RejectReason(id=3, desc='tau_time invalid'),
-    'nan': RejectReason(id=4, desc='contains NaN'),
+reject_reasons = {
+    'rms': Rejectreason(id=0, description='RMS invalid'),
+    'beam': Rejectreason(id=1, description='beam invalid'),
+    'bright_source': Rejectreason(id=2, description='bright source near'),
+    'tau_time': Rejectreason(id=3, description='tau_time invalid'),
+    'nan': Rejectreason(id=4, description='contains NaN'),
+    'flat': Rejectreason(id=5, description='data is flat'),
 }
 
-query_reject = """\
-INSERT INTO rejection
-  (image
-  ,rejectreason
-  ,comment
-  )
-VALUES
-  (%(imageid)s
-  ,%(reason)s
-  ,'%(comment)s'
-  )
-"""
 
-query_unreject = """\
-DELETE
-  FROM rejection
- WHERE image=%(image)s
-"""
+def sync_rejectreasons(session):
+    """
+    Check if rejectreasons are in sync. If not, insert as needed and commit.
 
-query_isrejected = """\
-SELECT rejectreason.description, rejection.comment
-  FROM rejection, rejectreason
- WHERE rejection.rejectreason = rejectreason.id
-   AND rejection.image = %(imageid)s
-"""
+    Args:
+        session (sqlalchemy.orm.Session): Database session.
+    """
+    if session.query(Rejectreason).count() != len(reject_reasons):
+        dbreason_id_rows = session.query(Rejectreason.id).all()
+        dbreason_ids = [row[0] for row in dbreason_id_rows]
+        for r in reject_reasons.values():
+            if r.id not in dbreason_ids:
+                logger.info("Added to database Rejectreason {}:'{}'".format(
+                    r.id, r.description
+                ))
+                session.add(r)
+        session.commit()
```

Comment:
hm jah, this is not really a long term solution but for now it works. We can't build manual logic for all schema changes.

This is a bit of a weird one, since the schema doesn't change, only the content. But in my opinion we should still manage that with SQLAlchemy & Alambic.